### ISSUE TITLE
TFlearn regression maybe implemented incorrectly - looks wrong

### ISSUE THREAD

**Issue by tiag01** [tflearn/tflearn#383]
IMHO that this also relates to #343, #378 and a few others.

I am working on converting a stacked LSTM tensor flow example to use tflearn:  https://github.com/tgjeon/TensorFlow-Tutorials-for-Time-Series/blob/master/lstm-for-epf.py.  It doesn't work with the latest Tensorflow, some of the contrib functions/libraries are no longer available and such.

I tried implement the corresponding network in tflearn and digging deeper into the network I found a bug with TFlearn regression implementation.  It's not actually implemented. @aymericdamien implemented a tensorflow example of regression https://github.com/aymericdamien/TensorFlow-Examples/blob/master/examples/2_BasicModels/linear_regression.py: 

`# Construct a linear model`
`pred = tf.add(tf.mul(X, W), b)`

`# Mean squared error`
`cost = tf.reduce_sum(tf.pow(pred-Y, 2))/(2*n_samples)`
`# Gradient descent`
`optimizer = tf.train.GradientDescentOptimizer(learning_rate).minimize(cost) `

Well looking at the simple regression example with tflearn at https://github.com/tflearn/tflearn/blob/master/examples/basics/linear_regression.py  you get a completely different graph - devoid of weights, biases and formula. You get just y. 

Please find the pictures below - one with regression that works with tensorflow and the other with tflearn.  I suspect @aymericdamien could fix this in a jiffy by putting in weights and biases and the normal formula for linear regression. Just to help out, there is a function in `from tensorflow.python.ops import nn`, that has `nn.xw_plus_b(tensor_in, weights, biases)`.

The non-working version:
<img width="811" alt="screen shot 2016-10-07 at 8 03 52 pm" src="https://cloud.githubusercontent.com/assets/53123/19209944/2e8e2fd4-8cca-11e6-8118-443a79be108f.png">

The working version:
<img width="697" alt="screen shot 2016-10-07 at 7 59 35 pm" src="https://cloud.githubusercontent.com/assets/53123/19209941/285d7e12-8cca-11e6-82c2-0b00fda143e4.png">

Thanks in advance,
Andrew

**Comment by aymericdamien** [comment]
I am not sure what is not working there? Does your training loss diverge in one case but not the other?

**Comment by tiag01** [comment]
I found the issue, my result was orders of magnitude with the google LSTM tensorflow code. I put in `model.fit(..., shuffle=False)` and  it's now reporting the right values.  Phew.
